### ISSUE TITLE
KeyboardInterrupt while Matplotlib is displaying ≥ 2 figures causes X server crash

### ISSUE THREAD

**Issue by Rufflewind** [matplotlib/matplotlib#7930]
### Bug report

**Bug summary**

  - When Matplotlib shows two or more windows (figures), hitting Ctrl+C in the terminal will kill Matplotlib (as intended), but also bring the entire X desktop environment crashing down.  The error logged in systemd/journalctl is:

        org.a11y.atspi.Registry[1907]: XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":0"
        org.a11y.atspi.Registry[1907]:       after 331 requests (331 known processed) with 0 events remaining.

**Code for reproduction**

- A minimum code snippet required to reproduce the bug, also minimizing the number of dependencies required

```python
import matplotlib
# doesn't seem to matter which, as long as it's interactive
#matplotlib.use("Qt5Agg")
import matplotlib.pyplot as plt
fig1, ax1 = plt.subplots()
ax1.plot([1, 2], [3, 4])
fig2, ax2 = plt.subplots()
ax2.plot([1, 2], [3, 4])
plt.show()
# Then, hit Ctrl+C without closing any of the windows.
```

**Actual outcome**

- The output produced by the above code, which may be a screenshot, console output, etc.

```
<all windows vanish>
<then screen turns black>
<and you are back on the login screen>
```

**Expected outcome**

- A description of the expected outcome from the code snippet
- If this used to work in an earlier version of Matplotlib, please note the version it used to work on

**Matplotlib version**

- Matplotlib version, Python version and Platform (Windows, OSX, Linux ...)
- How did you install Matplotlib and Python (pip, anaconda, from source ...)

I had two matplotlibs installed, one from Git (2.0.0rc2+2914.g1fa4dd705.dirty) and is installed for Python 3.6, the other is 1.5.3 packaged by the official Arch Linux repository and is installed for Python 2.7.  Could reproduce the problem on both.

Xorg version is 1.19.1.  The problem affects GtkAgg, Qt5Agg, and WxAgg.  Did not test Qt4Agg, but I bet it crashes too.

**Comment by anntzer** [comment]
I cannot reproduce the issue on Arch Linux in any of the following configs -- all with xorg-server 1.19.1:
- py3.6 (from official repos) mpl2.0.0rc2+2914 qt5agg
- py2.7 mpl1.5.3 (from official repos) gtkagg OR qt5agg OR wxagg.

**Comment by Rufflewind** [comment]
Forgot to mention, I'm also using XMonad 0.12 as the window manager.

**Comment by tacaswell** [comment]
Can you reproduce this without mpl, just creating a Qt window and showing it? 

I assume you are running this as `python script.py`?

**Comment by anntzer** [comment]
See https://bbs.archlinux.org/viewtopic.php?id=222066?

**Comment by Rufflewind** [comment]
Sorry, it appears the [problem is not related to matplotlib](https://github.com/xmonad/xmonad/issues/70).  Rather, it's a bug with the WM (exposed by recent changes to Xorg?).  I ran into the Arch Linux thread just now, which @anntzer also linked.

It seems that examining journalctl and Xorg.log was insufficient.  I had to manually redirect the stderr of `startx` to a file to see the *real* error:

    xmonad-x86_64-linux: user error (Error in function getWindowAttributes)

That is definitely an error in Haskell-land.  Thanks all for the assistance!
